src/Nix/Builtins.hs

```diff
@@ -913,9 +939,11 @@ hashString algo s = Prim $ do
         _ -> throwError $ ErrorCall $ "builtins.hashString: "
             ++ "expected \"md5\", \"sha1\", \"sha256\", or \"sha512\", got " ++ show algo
 
+-- TODO Double-check this
 placeHolder :: MonadNix e m => m (NValue m) -> m (NValue m)
-placeHolder = fromValue @Text >=> \_ -> do
-    h <- runPrim (hashString "sha256" "fdasdfas")
+placeHolder = fromValue >=> fromStringNoContext >=> \ns -> do
+    h <- runPrim (hashString (principledMakeNixStringWithoutContext "sha256")
+                             (principledMakeNixStringWithoutContext ns))
```

Comment:
Should be an error if `ns` has context.

Reply:
The `fromStringNoContext` function does error.  The name `ns` is misleading because it actually has type `Text`.  I'll fix the name.